### PR TITLE
Camunda 7.20 and Spring Boot 3 upgrade, includes fixes for conditional logic that caused bugs previously.

### PR DESCRIPTION
**Pull request by MartyFox**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-15649

### Change description ###
Camunda 7.20 upgrade, Spring Boot 3 upgrade and relevant changes for both.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -141,44 +141,42 @@ sonarqube {
...
+    dependencies {
+        imports {
+            // Before upgrading, please copy relevant migration scripts in src/main/resources/db/migration/ from https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/resources/org/camunda/bpm/engine/db/upgrade/
+            mavenBom 'org.camunda.bpm:camunda-bom:7.20.3-ee'
+        }
     }
-  }
 }
 
+ext['spring-security.version'] = '6.2.2'
```
Comment (timja) [inline build.gradle:152]: this is probably not needed for a minimal change, it just fixed the `debug` attribute not working.

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java [hunk 2]
```diff
@@ -6,21 +6,30 @@
...
 @Configuration
 @ConditionalOnProperty(prefix = "camunda.ui.auth", name = "enabled", matchIfMissing = false)
 @EnableWebSecurity
 @Order(100)
-public class WebSecurityWebAppConfig extends WebSecurityConfigurerAdapter {
+public class WebSecurityWebAppConfig {
+
+    @Scope("prototype")
+    @Bean
+    MvcRequestMatcher.Builder mvc(HandlerMappingIntrospector introspector) {
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java:30]: I assume this is needed for something?

**Review by timja — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java (hunk 2 above)
Comment (MartyFox) [reply src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/WebSecurityWebAppConfig.java:30]: Yes reverting back to the normal Spring Security oAuth requires this for some reason, fails without it.
Spring Cloud Azure had a similar issue with a different resource that just needed to be declared.

File: build.gradle (hunk 1 above)
Comment (MartyFox) [reply build.gradle]: removed

**Pull request merged**